compiler/cpp/test/analyzer/AnalyzerTest.cpp

```diff
@@ -0,0 +1,118 @@
+                "Incompatible data types: '\"Test string\"' is not a boolean"
+        },
+        {
+                "conditional-section-error-string-int", // Expects string, but data has int
+                false,
+                "Incompatible data types: '1234' is not a string"
+        },
+        {
+                "conditional-section-error-string-boolean", // Expects string, but data has boolean
+                false,
+                "Incompatible data types: 'true' is not a string"
+        },
+        {
+                "single-statement-valid",
+                true,
+                ""
+        },
+        {
+                "conditional-section-error-int-string", // Expects int, but data has string
+                true,
+                "Incompatible data types: '\"Test string\"' is not an integer"
+        },
+        {
+                "conditional-section-error-int-boolean", // Expects int, but data has boolean
+                true,
+                "Incompatible data types: 'false' is not an integer"
+        },
+        {
+                "conditional-section-error-boolean-int", // Expects boolean, but data has int
+                true,
+                "Incompatible data types: '4321' is not a boolean"
+        },
+        {
+                "conditional-section-error-boolean-string", // Expects boolean, but data has string
+                true,
+                "Incompatible data types: '\"Test string\"' is not a boolean"
+        },
+        {
+                "conditional-section-error-string-int", // Expects string, but data has int
+                true,
+                "Incompatible data types: '1234' is not a string"
+        },
+        {
+                "conditional-section-error-string-boolean", // Expects string, but data has boolean
+                true,
+                "Incompatible data types: 'true' is not a string"
+        }
+};
+
+class AnalyzerTests : public ::testing::TestWithParam<AnalyzerParams> {
+};
+
+TEST_P(AnalyzerTests, TestAnalyzerWithValidAndInvalidTestFiles) {
+    AnalyzerParams param = GetParam();
+    // Load parser input and expected output
+    FileReader reader = FileReader("test-files");
+    std::string fileInput = reader.fileToString("analyzer/" + param.fileName, "input-file.txt");
+    std::string dataInput = reader.fileToString("analyzer/" + param.fileName, "input-data.json");
+    // Initialize analyzer
+    JSONParser jsonParser = JSONParser(dataInput);
+    Analyzer analyzer = Analyzer(false, fileInput, jsonParser, "//", "/*", "*/");
+    // Execute test case
+    try {
+        analyzer.executeAnalysis();
+        if (param.errorMessage.length() > 0)
+            FAIL() << "Expected error message '" + param.errorMessage + "', but got no error";
+        SUCCEED();
+    } catch (IncompatibleTypesException &error) {
+        if (param.errorMessage.length() == 0)
+            FAIL() << "Expected no error, but got '" + std::string(error.what()) + "'";
```

Comment:
"Expected no error, but got...", however the if statement is checking if there is no error message `param.errorMessage.length() == 0`.

Vice versa holds true for line 103.

---

Ah ok, got it now, the `param.errorMessage` is the error message that we expect...

Reply:
Exactly